src/lib.rs

```diff
@@ -16,23 +17,22 @@ const ESCAPE_KEY: &str = "Escape";
 
 const STORAGE_KEY: &str = "todos-seed";
 
-// ------ Url path parts ------
-const ACTIVE: &str = "active";
-const COMPLETED: &str = "completed";
-
+add_router!();
```

Comment:
Why we need it?

Reply:
it is adding this code

```rust
#[proc_macro]
pub fn add_router(_item: TokenStream) -> TokenStream {
    "thread_local! {
    pub(crate) static ROUTER: Router<Route> = Router::new();
}

/// Access the router from global state
fn router() -> Router<Route,> {
    ROUTER.with(Clone::clone,)
}"
    .parse()
    .unwrap()
}
```

is it a bad strategy to do this macro for saving so complexity?

Reply:
It looks like that code could written directly in routing crate without macros

Reply:
Yes it is true. I remember I did the same for Urls at some point. I am gonna try this.

Reply:
I will look at it better tomorrow. I got issue with the typing because we need to routes defined in the application that we do build .

Reply:
Yeah, I see the problem now. And `router::<Route>()` instead of `router()` wouldn't be too nice, so it'll take some time to resolve this problem.

Reply:
One thing could be to add this to init instead

```rust
#[derive(Router)]
fn init(url: Url, orders: &mut impl Orders<Msg>) -> Model {

    router()
        .init(url)
        .subscribe(orders.subscribe_with_handle(
        |subs::UrlRequested(requested_url, _)| router().confirm_navigation(requested_url),
    ));

    Model {
        todos: LocalStorage::get(STORAGE_KEY).unwrap_or_default(),
        new_todo_title: String::new(),
        selected_todo: None,
    }
}

```

It feels wierd to have this on the init function for something that will be there for the entire application.

I could also put it inside RoutingModules. But it also feels weird. Maybe the `add_router()!` is enough for now ? I think there are more serious and important error/wrong concepts to fix in other part of the router.